### PR TITLE
rollout.py: check whether rollout matches latest release

### PR DESCRIPTION
**Pull request by jlebon**
This is an easy way to sanity-check that the build IDs match up. The
version we rollout should almost always match what the latest release on
that stream is. Add a check for this.

So... this script now does networking. Which should be fine in most
cases, but add a switch in case a user doesn't want to depend on the
network.

### CODE REVIEW

File: rollout.py [hunk 1]
```diff
@@ -99,6 +104,15 @@ def report(info):
         print(f"latest entry {version} on {stream} is not a rollout")
         return
 
+    latest_info = "unvalidated"
+    if not skip_build_id_check:
+        releases_url = RELEASES.substitute(stream=stream)
+        releases = requests.get(releases_url).json()["releases"]
+        if releases[-1]["version"] == version:
+            latest_info = "latest"
+        else:
```
Comment (bgilbert) [inline rollout.py:113]: As an extra feature, it'd be straightforward to distinguish between stale and bogus versions here.

File: rollout.py [hunk 2]
```diff
@@ -170,6 +184,8 @@ def _main():
             help='rollout start (e.g. "10 am")')
     add.add_argument('duration', metavar='duration-hours', type=int,
             help='rollout duration (e.g. "48")')
+    add.add_argument('--skip-build-id-check', action='store_true',
+                     help='skip validiting build IDs')
```
Comment (bgilbert) [inline rollout.py:188]: `validating`

Hmm, the rest of the code calls these `versions`.  We should probably be consistent; which name should we use?

**Review by bgilbert — COMMENTED** [review]
Nice!

**Comment by jlebon** [comment]
Good feedback, thanks! Updated!

**Review by travier — APPROVED** [review]
Nice! LGTM (but untested)

**Review by bgilbert — APPROVED** [review]

**Pull request merged**